ocaml/idl/datamodel_host.ml

```diff
@@ -2120,6 +2138,10 @@ let t =
         ; field ~qualifier:DynamicRO ~lifecycle:[] ~ty:(Set update_guidances)
             "recommended_guidances" ~default_value:(Some (VSet []))
             "The set of recommended guidances after applying updates"
+        ; field ~qualifier:DynamicRO ~lifecycle:[] ~ty:up_to_date_state
```

Comment:
Could this be typed as `Option Bool`?

Reply:
Yes, I think it can, but when getting it, you will have to match it with None to do a translation like:

      ; make_field ~name:"latest-synced-updates-applied"
          ~get:(fun () ->
             match  (x ()).API.host_latest_synced_updates_applied with
             | Some a -> Bool.to_string a
             | None -> "unknown"
          )
          ()
      ]

While with a new type like:

let latest_synced_updates_applied_state =
  Enum
    ( "latest_synced_updates_applied_state"
    , [
        ( "yes"
        , "The host is up to date with the latest updates synced from remote \
           CDN" 
        )
      ; ( "no"
        , "The host is outdated with the latest updates synced from remote CDN" 
        )
      ; ( "unknown"
        , "If the host is up to date with the latest updates synced from \
           remote CDN is unknown"
        )
      ]
    )

The definition of all the state is in one place, I think this is easier to understand and would like to keep it.